backend/apps/ifc_validation/checks/header_policy/mvd_parser.py

```diff
@@ -0,0 +1,102 @@
+from lark import Lark, Transformer
+import ifcopenshell
+
+
+# https://standards.buildingsmart.org/documents/Implementation/ImplementationGuide_IFCHeaderData_Version_1.0.2.pdf
+mvd_grammar = r'''
+    start: entry+
+
+    entry: "ViewDefinition" "[" simple_value_list "]"   -> view_definition
+         | "Comment" "[" comment_text "]" -> comment
+         | GENERIC_KEYWORD "[" value_list_set "]" -> dynamic_option
+
+    %declare COMMENT_TEXT  // Ensure Lark treats it with higher priority
+
+    GENERIC_KEYWORD: /[A-Za-z0-9_]+/
+    
+    simple_value_list: value ("," value)*
+    
+    value_list_set: value_set (";" value_set)*
+
+    value_set: set_name ":" simple_value_list
+    
+    set_name: /[A-Za-z0-9_]+/
+    
+    value: /[A-Za-z0-9 _-]+/
+    
+    comment_text: /[^\[\]]+/  
+
+    %import common.WS
+    %ignore WS
+'''
+
+
+parser = Lark(mvd_grammar, parser='lalr')
+class DescriptionTransform(Transformer):
+    def __init__(self):
+        self.mvd = []
+        self.keywords = set()
+        self.comments = ""
+        self.exchangerequirement = ""
+        self.option = ''
+
+
+    def view_definition(self, args):
+        self.keywords.add('mvd')  
+        self.mvd.extend(args[0])  
+
+
+    def dynamic_option(self, args):
+        """
+        e.g. in case of 'Remark' as optional keyword in the description
+        The value can be retrieved through DescriptionTransform.remark
+        """
+        key = str(args[0]).lower()
+        attr_name = f"{key}"
+        if attr_name not in self.keywords:
+            setattr(self, attr_name, {})
+            self.keywords.add(attr_name)
+        dynamic_dict = getattr(self, attr_name)
+        for value_set in args[1]:
+            set_name, *values = value_set
+            dynamic_dict[set_name] = values if len(values) > 1 else values[0]
+
+
+    def comment(self, args):
+        self.keywords.add('comment')
+        self.comments = " ".join(str(child) for child in args[0].children).strip()
+
+    def simple_value_list(self, args):
+        return [str(arg) for arg in args]
+
+    def value_list_set(self, args):
+        return args
+
+    def value_set(self, args):
+        return [str(args[0])] + args[1]
+
+    def value(self, args):
+        return str(args[0])
+
+    def set_name(self, args):
+        return str(args[0])
+    
+    @property 
+    def other_keywords(self):
+        """"
+        The predefined keywords are 'ViewDefinition', 'Option', 'Comment', 'ExchangeRequirement'
+        Keywords in the description not from this lists are returned
+        """
+        return {k for k in self.keywords if k not in {'mvd', 'comment', 'exchangerequirement', 'option'}}
+
+
+def parse_mvd(text):
+    print('parsing ...')
+    parsed_description = DescriptionTransform()
+    try:
+        parse_tree = parser.parse(text)
+        parsed_description.transform(parse_tree)
+    except:
```

Comment:
Please never generic try-except. Only catch the exceptions you expect.

Reply:
[I've made it more explicit ](https://github.com/buildingSMART/validate/pull/146/commits/2686d7a038924dd049335de5d2e36c00378820bd)

Also these changes are modified in ifcopenshell
https://github.com/IfcOpenShell/IfcOpenShell/pull/6154/commits/7e4bb2094e25609f01c795d24678560701638af4